docs/src/quickstart.md

```diff
@@ -85,7 +85,10 @@ Turing.@model turing_model(
     τ ~ truncated(Cauchy(0, 5), 0, Inf)
     θ = TV(undef, J)
     θ .~ Normal(μ, τ)
-    y ~ MvNormal(θ, σ)
+    for i in eachindex(y)
```

Comment:
Nah, not faster. But for `elementwise_loglikelihood` it's not clear what `MvNormal` in this case means. Is `y` a single observation or is it a vector of observations? And so if you want to use `elementwise_loglikelihood`, you need to be "explicit" about which you mean in the model definition, unfortunately :confused:

Reply:
Ah, my question was more about `for i in eachindex(y)` vs `for i in 1:J`, since `J` is a passed parameter that is assumed to be the length of `y` and `σ`. IIRC at some time in the past, the advice in Turing was to explicitly have size parameters like `J` when possible because of some efficiency boost.

Reply:
That makes sense though about `elementwise_loglikelihood`.

Reply:
Ah, uhh, I don't think so? The only reason why I could see that being the case is due to constant propagation? Other than that, I'm not sure if there's going to be much of a difference. I think you might know more about this than me, but could it make a difference for AD if the number of iterations for the for-loop is known at compile time?

Reply:
Yeah, I don't think so. If `J` is a passed parameter, I don't think its value would be known at compile time anyways. And in fact if `y` is statically sized, then `eachindex(y)` maybe could cause the loop to be unrolled. I don't think either case necessarily is better for AD.

Reply:
> If J is a passed parameter, I don't think its value would be known at compile time anyways.

The evaluator for the model is a closure created when you instantiate the model, so at least in theory `J` is known at compile time (whether or not it is indeed treated as a static variable by the Julia compiler I'm not sure, though I would expect so).

> And in fact if y is statically sized,

Yeah I was thinking that too.